### PR TITLE
feature/AKD-150: Added Azure 'VM Managed Disks Enabled' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/virtualmachines/vmManagedDisks.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                return rcb();
+            }
+
+            if (!virtualMachines.data.length) {
+                helpers.addResult(results, 0, 'No existing Virtual Machines found', location);
+                return rcb();
+            }
+
+            for (let virtualMachine of virtualMachines.data) {
+                if (virtualMachine.storageProfile && virtualMachine.storageProfile.osDisk &&
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/vmManagedDisks.js:35]: We should also check whether data disk type is also managed

File: plugins/azure/virtualmachines/vmManagedDisks.spec.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        'name': 'test-vm',
+        'id': '/subscriptions/123/resourceGroups/AQUA-RESOURCE_GROUP/providers/Microsoft.Compute/virtualMachines/test-vm',
+        'type': 'Microsoft.Compute/virtualMachines',
+        'storageProfile': {
+            'osDisk': {
+                'osType': 'Linux',
+                'name': 'test-vm-disk',
+                'managedDisk': {
+                    'storageAccountType': 'Premium_LRS',
+                    'id': '/subscriptions/dce7d0ad-ebf6-437f-a3b0-28fc0d22117e/resourceGroups/Ali-Recource_Group/providers/Microsoft.Compute/disks/test-vm-disk'
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/vmManagedDisks.spec.js:33]:
```suggestion
                    'id': '/subscriptions/123/resourceGroups/AQUA-RESOURCE_GROUP/providers/Microsoft.Compute/disks/test-vm-disk'
```

File: plugins/azure/virtualmachines/vmManagedDisks.spec.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                },
+                'diskSizeGB': 30
+            },
+            'dataDisks': [
+                {
+                    'name': 'test-vm-data-disk',
+                    'managedDisk': {
+                        'storageAccountType': 'Standard_LRS',
+                        'id': '/subscriptions/dce7d0ad-ebf6-437f-a3b0-28fc0d22117e/resourceGroups/ALI-RECOURCE_GROUP/providers/Microsoft.Compute/disks/test-vm-data-disk'
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/vmManagedDisks.spec.js:42]:
```suggestion
                        'id': '/subscriptions/123/resourceGroups/AQUA-RESOURCE_GROUP/providers/Microsoft.Compute/disks/test-vm-data-disk'
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualmachines/vmManagedDisks.js (hunk 1 above)
Comment (ali-imran7) [reply plugins/azure/virtualmachines/vmManagedDisks.js:35]: @AkhtarAmir We cannot attach an unmanaged data disk if VM is using managed disks and vice versa. So checking OS disk only will ensure that what kind of disks VM is using.

**Review by giorod3 — APPROVED** [review]

**Pull request merged**